app/src/main/java/com/github/fribourgsdp/radio/data/view/UserProfileActivity.kt

```diff
@@ -200,7 +206,8 @@ open class UserProfileActivity : MyAppCompatActivity(), KeepOrDismissPlaylistDia
             user.name = it.result.name
             user.id = it.result.id
             user.isGoogleUser = true
-            db.setUser(user.id,user)
+            val onlineUser = user.onlineCopy()
+            db.setUser(onlineUser.id,onlineUser)
```

Comment:
lines 186-190 and 206-210 are same

Reply:
Even though these line are the same, I personally prefer to have each button having its own behavior written directly. I think it makes more sense semantically. What do you think?

Reply:
I see what you mean but I agree with @Flamantrose321 that this may be a better idea to make a method out of it (with `db` as a param) as those exact code snippets are also used in `PlaylistSongsFragment.kt` and `AddPlaylistActivity.kt`

Reply:
yes agree with @VictorSchneuwly

Reply:
I think I did what you asked for. Let me know if it is okay.